source/common/router/config_impl.cc

```diff
@@ -103,7 +137,8 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost, const Json:
   }
 }
 
-bool RouteEntryImplBase::matches(const Http::HeaderMap& headers, uint64_t random_value) const {
+const Route* RouteEntryImplBase::matches(const Http::HeaderMap& headers,
```

Comment:
There is no point in having this method (the base matches method) return a Route*. Just have it return bool like it used to (and change method name to something else so it's not an override).